Source/FlowEditor/Private/Graph/FlowGraphEditor.cpp

```diff
@@ -1269,4 +1310,220 @@ bool SFlowGraphEditor::CanJumpToNodeDefinition() const
+                Graph->Modify();
+            	
+                UFlowNode_NamedRerouteDeclaration* Declaration = CastChecked<UFlowNode_NamedRerouteDeclaration>(
+                    FFlowGraphSchemaAction_NewNode::CreateNode(
+                        GetCurrentGraph(), nullptr, UFlowNode_NamedRerouteDeclaration::StaticClass(),
+                        FVector2D(GraphNode->NodePosX + 100, GraphNode->NodePosY))->GetFlowNodeBase());
+                
+                if (!Declaration)
+                {
+                    return;
+                }
+            	
+                UEdGraphPin* DeclarationOutputPin = nullptr;
+                for (auto* Pin : Declaration->GetGraphNode()->GetAllPins())
+                {
+                    if (Pin->Direction == EEdGraphPinDirection::EGPD_Output)
+                    {
+                        DeclarationOutputPin = Pin;
+                        break;
+                    }
+                }
+                if (!ensure(DeclarationOutputPin))
+                {
+                    return;
+                }
+            	
+                UEdGraphPin* RerouteOutputPin = GraphNode->OutputPins[0];
+                for (auto* OutputPin : RerouteOutputPin->LinkedTo)
+                {
+                    DeclarationOutputPin->MakeLinkTo(OutputPin);
+                }
+            	
+                constexpr int32 OffsetX = -100;
+                int32 OffsetY = -50;
+                for (auto* InputPin : GraphNode->InputPins[0]->LinkedTo)
+                {
+                    UFlowNode_NamedRerouteUsage* Usage = CastChecked<UFlowNode_NamedRerouteUsage>(
+                        FFlowGraphSchemaAction_NewNode::CreateNode(
+                            GetCurrentGraph(), nullptr, UFlowNode_NamedRerouteUsage::StaticClass(),
+                            FVector2D(GraphNode->NodePosX + OffsetX, GraphNode->NodePosY + OffsetY))->GetFlowNodeBase());
+                    
+                    if (!Usage)
+                    {
+                        continue;
+                    }
+                	
+                    UEdGraphPin* UsageInputPin = nullptr;
+                    for (auto* Pin : Usage->GetGraphNode()->GetAllPins())
+                    {
+                        if (Pin->Direction == EEdGraphPinDirection::EGPD_Input)
+                        {
+                            UsageInputPin = Pin;
+                            break;
+                        }
+                    }
+                    if (!ensure(UsageInputPin))
+                    {
+                        continue;
+                    }
+                	
+                    InputPin->MakeLinkTo(UsageInputPin);
+                    Usage->RegisterLinkedDeclaration(Declaration);
+                    OffsetY += 100; 
+                }
+            	
+                GraphNode->DestroyNode();
+                SetNodeSelection(Declaration->GetGraphNode(), true);
```

Comment:
Maybe it would be good to select also all created Usage nodes? Currently only last Usage node and Declaration node are selected.  (FFlowGraphSchemaAction_NewNode::CreateNode clears the selection set if `bSelectNewNode == true` and it is defaulted to `true`, that is why last inserted Usage node is selected)

Reply:
Yes, I think it would be good to select all the new usages. I will start working on the new commit with the necessary changes, also, I think it is not necessary to wrap #if WITH_EDITOR to the RegisterLinkedDeclaration() and UnregisterLinkedDeclaration() functions, what do you think about that?